core-bundle/contao/widgets/SerpPreview.php

```diff
@@ -41,7 +41,7 @@ public function generate()
 		}
 
 		$id = $model->id;
-		$title = StringUtil::substr($this->getTitle($model), 64);
+		$title = StringUtil::substr(str_replace(array('&nbsp;', '&shy;'), array(' ', ''), $this->getTitle($model)), 64);
```

Comment:
Or maybe also just replace it with the unicode chars here too:
```suggestion
		$title = StringUtil::substr(str_replace(array('&nbsp;', '&shy;'), array("\u{A0}", "\u{AD}"), $this->getTitle($model)), 64);
```

Reply:
I am pretty sure Google removes any given `&nbsp;` or `&shy;`.

Running `for (const txt of document.querySelectorAll('h3')) {console.log(txt)}` in the js-console on any google search result page never returned any `&nbsp;` or `&shy;`. Even searching with a long word didn't return word breaks:
https://www.google.com/search?q=donaudampfschifffahrtselektrizit%C3%A4tenhauptbetriebswerkbauunterbeamtengesellschaft
They also strip these entities from the page description – just compare the SERP source code of above search with the entry in wikipeadia (with a lot of `&shy;`s!

So I would just strip `&shy;` and replace `&nbsp;` with a space character.

Reply:
I don’t see any entities in the title there on wikipedia: 
![Bildschirmfoto 2022-12-13 um 17 02 35](https://user-images.githubusercontent.com/367169/207383130-78d21ee8-1e7a-4edb-9de0-813f032d0523.png)

Reply:
> I don’t see any entities in the title there on wikipedia

I was referring to the description. This is a screenshot from the search result:
![image](https://user-images.githubusercontent.com/3359867/207391015-52735909-230d-4c2a-a419-70774f57002c.png)

And this is the text on the wikipedia page used by google (wikipedia doesn't use the description meta tag):
![image](https://user-images.githubusercontent.com/3359867/207390879-61847fbb-fd14-4ea4-9c9e-3bf4db901aa4.png)
As one can see, google also strips the `<b>` tag…